README.rst

```diff
@@ -40,6 +40,18 @@ UCI configuration options must go in ``/etc/config/monitoring``.
 - ``monitored_interfaces``: interfaces that needs to be monitored, defaults to ``*`` for all interfaces.
 - ``interval``: time after which device data should be sent to server, defaults to ``300``.
 - ``verbose_mode``: can be enabled (set to ``1``) to ease `debugging <#debugging>`__ in case of issues, defaults to ``0`` (disabled).
+- ``required_memory``: available memory required to save data temporarily, defaults to ``0.05``.
+- ``max_retries``: number of time the service should try to send data to server in case of failure, defaults to ``5``.
+
+OpenWISP Monitoring Agent
+-------------------------
+
+We use two procd services in `monitoring agent <https://github.com/openwisp/openwrt-openwisp-monitoring/blob/master/openwrt-openwisp-monitoring/files/monitoring.agent>`_, one for collecting the data and other for sending the data.
+
+**SIGUSR1** signals are used to instantly send the data when collected. However, the service will keep trying
+to send data periodically.
+
+**Note:-** It may be possible that the server is down for a long time, then sending data may take time then the interval after which data is collected creating a gap in data points. To avoid this, two services are used.
```

Comment:
Citing the server as being down is not a great example, the most common case is the OpenWRT device simply not being able to send the data for network connectivity issues.

I think we could divide this section further and explain what happens in the most important cases, eg:

1st heading: collecting vs sending

Intro: this agent uses 2 different processes, one for sending data and one for collecting it.
This allows to handle failures in sending the data in a more flexible way because old data stored during network connectivity issues can be sent at the same time while new data is being collected: if old data has piled up for sending and takes several minutes to be uploaded, new data is collected without being delayed by the sending.

Subheading 1: Collect mode

In this mode the agent is in charge of collecting data ...
The data is collected only if ...
If there is not enough memory ...
Once data is collected, a signal is sent to the other agent so that ...

Subheading 2: Send mode

In this mode the agent .....
If sending fails .... explain retries...
If all attempts fail ...